frontend/src/components/modals/AddCustomerModal.tsx

```diff
@@ -0,0 +1,250 @@
+                  step={stepData.step}
+                  key={stepData.step}
+                  description={stepData.description}
+                  onClick={() => setStep(stepData.step)}
+                />
+              ))}
+            </div>
+          )}
+          {step === 1 && (
+            <SelectCustomerInfo
+              name={formValues.name}
+              onNameChange={(value) =>
+                setFormValues({ ...formValues, name: value })
+              }
+              email={formValues.email}
+              onEmailChange={(value) =>
+                setFormValues({ ...formValues, email: value })
+              }
+              colorType={colorType}
+              onColorTypeChange={(value) => setColorType(value)}
+              color={formValues.color}
+              onColorChange={(value) =>
+                setFormValues({ ...formValues, color: value })
+              }
+            />
+          )}
+          {step === 2 && (
+            <SelectRepresentatives
+              representatives={representatives}
+              onRepresentativeChange={(idx, checked) => {
+                const copy = [...(representatives ?? [])];
+                if (!copy) return;
+                copy[idx].checked = checked;
+                setRepresentatives(copy);
+              }}
+            />
+          )}
+          {step === 3 && (
+            <div className={classes(css.leftText)}>
+              <b>All done.</b> You can now confirm adding the client, or go back
+              to previous steps if you missed something.
+            </div>
+          )}
+          {step === 4 && (
+            <div className={classes(css.modalDoneContent)}>
+              <CheckIcon />
+              <h6>Awesome! {formValues.name} is now added!</h6>
+            </div>
+          )}
+          <Alert
+            show={errorMessage.length > 0}
+            variant="danger"
+            dismissible
+            onClose={() => setErrorMessage('')}
+          >
+            {errorMessage}
+          </Alert>
+        </ModalContent>
+        <ModalFooter>
+          <ModalFooterButtonDiv empty={step === 4}>
+            {step < 4 && (
+              <button
+                className="button-large cancel"
+                onClick={handleCancel}
+                type="button"
+              >
+                {step === 0 ? (
+                  <Trans i18nKey="No, go back" />
+                ) : (
+                  <Trans i18nKey="Cancel" />
+                )}
+              </button>
+            )}
+          </ModalFooterButtonDiv>
+          <ModalFooterButtonDiv>
+            {isLoading ? (
+              <LoadingSpinner />
+            ) : (
+              <button className="button-large" type="submit">
+                {step === 0 && <Trans i18nKey="Yes, I want to cancel" />}
+                {step > 0 && step < 4 && <Trans i18nKey="Confirm" />}
+                {step === 4 && <Trans i18nKey="Great!" />}
+              </button>
```

Comment:
Confirm-näppäin on figmassa harmaana silloin, kun kaikkia vaadittavia kenttiä ei ole täytetty